OmniSci GroupBy tests fail with OmniSci 5.8.0

```
FAILED modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py::TestGroupby::test_series_astype
FAILED modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py::TestGroupby::test_df_astype
FAILED modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py::TestGroupby::test_taxi_q4[True]
FAILED modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py::TestGroupby::test_taxi_q4[False]
FAILED modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py::TestGroupby::test_taxi_q4[None]
```

@dchigarev , take a look, please.

--- a/modin/experimental/core/execution/native/implementations/omnisci_on_native/expr.py
+++ b/modin/experimental/core/execution/native/implementations/omnisci_on_native/expr.py
@@ -219,6 +219,11 @@ class BaseExpr(abc.ABC):
         BaseExpr
             The cast expression.
         """
+        # From float to int cast we expect truncate behavior but CAST
+        # operation would give us round behavior.
+        if is_float_dtype(self._dtype) and is_integer_dtype(res_type):
+            return self.floor()
+
         new_expr = OpExpr("CAST", [self], res_type)
         return new_expr
 

